Parse/PFAuthenticationProvider.h

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <Bolts/BFTask.h>
+
+#import <Parse/PFConstants.h>
+
+PF_ASSUME_NONNULL_BEGIN
+
+/*!
+ Provides a general interface for delegation of third party authentication with <PFUser>s.
+ */
+@protocol PFAuthenticationProvider <NSObject>
+
+/*!
+ @returns Returns a unique name for the type of authentication the provider does.
+ */
++ (NSString *)authType;
+
+/*!
+ @abstract Deauthenticates (logs out) the user associated with this provider.
+
+ @returns A task that resolves to anything if deauthentication succeeded, otherwise - faulted task.
+ */
+- (BFTask *)deauthenticateInBackground;
```

Comment:
I am glad you asked. For both of these - we absolutely don't care about the result type of the task, so I would rather leave this as is.